doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+# Metrics
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/metrics.js -->
+
+The `node:metrics` module provides an API for application instrumentation and
+performance monitoring. It offers various metric types and built-in exporters
+for popular monitoring systems.
+
+The module can be accessed using:
+
+```mjs
+import * as metrics from 'node:metrics';
+```
+
+```cjs
+const metrics = require('node:metrics');
+```
+
+## Overview
+
+The metrics API enables developers to instrument their applications with custom
+metrics that can be collected and exported to monitoring systems. All metrics
+publish their data through the `node:diagnostics_channel` module, allowing for
+flexible consumption patterns.
+
+### Example
+
+```mjs
+import { counter, timer, statsdStream } from 'node:metrics';
+import { createWriteStream } from 'node:fs';
+
+// Create a counter metric
+const apiCalls = counter('api.calls', { service: 'web' });
+
+// Create a timer factory
+const requestTimer = timer('api.request.duration', { service: 'web' });
+
+// Export metrics to StatsD format
+const statsd = statsdStream();
+statsd.pipe(createWriteStream('metrics.log'));
+
+// Use metrics in your application
+function handleRequest(req, res) {
+  const timer = requestTimer.create({ endpoint: req.url });
+
+  apiCalls.increment();
+
+  // Process request...
+
+  timer.stop();
+}
+```
+
+```cjs
+const { counter, timer, statsdStream } = require('node:metrics');
```

Comment:
I'd prefer the constructor methods here to follow a naming pattern like `createCounter`, `createTimer`, etc, both for ergonomics and for consistency (see `createHistogram(...)` in `perf_hooks`.

```js
const counter = createCounter('api.calls', { service: 'web' });
const timer = createTimer('api.request.duration', { service: 'web' });
```

Reply:
My original thinking was to keep the names as simple as possible to be not overly verbose and fairly quick to read, but then in writing tests and docs for it all I found a common case was to want to name the variable the same thing as the function. I may be a bit too radically minimalist about the naming. 😅